processing/src/main/java/org/apache/druid/segment/filter/BoundFilter.java

```diff
@@ -106,22 +106,27 @@ public BitmapColumnIndex getBitmapColumnIndex(ColumnIndexSelector selector)
       }
       final NumericRangeIndex rangeIndex = indexSupplier.as(NumericRangeIndex.class);
       if (rangeIndex != null) {
-        final Number lower = boundDimFilter.hasLowerBound() ? Double.parseDouble(boundDimFilter.getLower()) : null;
-        final Number upper = boundDimFilter.hasUpperBound() ? Double.parseDouble(boundDimFilter.getUpper()) : null;
-        final BitmapColumnIndex rangeBitmaps = rangeIndex.forRange(
-            lower,
-            boundDimFilter.isLowerStrict(),
-            upper,
-            boundDimFilter.isUpperStrict()
-        );
-        if (rangeBitmaps != null) {
-          // preserve sad backwards compatible behavior where bound filter matches 'null' if the lower bound is not set
-          if (boundDimFilter.hasLowerBound() && !NullHandling.isNullOrEquivalent(boundDimFilter.getLower())) {
-            return rangeBitmaps;
-          } else {
-            return wrapRangeIndexWithNullValueIndex(indexSupplier, rangeBitmaps);
+        try {
+          final Number lower = boundDimFilter.hasLowerBound() ? Double.parseDouble(boundDimFilter.getLower()) : null;
+          final Number upper = boundDimFilter.hasUpperBound() ? Double.parseDouble(boundDimFilter.getUpper()) : null;
+          final BitmapColumnIndex rangeBitmaps = rangeIndex.forRange(
+              lower,
+              boundDimFilter.isLowerStrict(),
+              upper,
+              boundDimFilter.isUpperStrict()
+          );
+          if (rangeBitmaps != null) {
+            // preserve sad backwards compatible behavior where bound filter matches 'null' if the lower bound is not set
+            if (boundDimFilter.hasLowerBound() && !NullHandling.isNullOrEquivalent(boundDimFilter.getLower())) {
+              return rangeBitmaps;
+            } else {
+              return wrapRangeIndexWithNullValueIndex(indexSupplier, rangeBitmaps);
+            }
           }
         }
+        catch (NumberFormatException ignored) {
+          // bounds are not numeric?
+        }
```

Comment:
I too am wondering why we need this here, and if there's a better way we can handle it.

Reply:
I too am wondering why we need this here, and if there's a better way we can handle it.

Reply:
I added this as a reaction to adding the 'auto' types to the `BaseFilterTest`. Specifically, `FloatAndDoubleFilteringTest.testFloatColumnFilteringWithNonNumbers` throws a bunch of garbage at the bound filter and expects it to still work (instead of being some failure condition.. yet another reason to add a flag to replace bound filter with something more sql compatible).

To behave consistently when the number column has indexes with these tests, it either needs to fall back to predicate index, or use value matcher - either makes the same results.. and the cardinality check of `ColumnIndexSupplier.skipComputingPredicateIndexes` makes it so if the predicate filter matches too much it will be skipped and fall back to value matcher, so eating this exception here seemed like unfortunately the best option.

My preference would be to just like.. validate the bounds are legit for the comparison type up front, but thats a behavior change, so maybe if i ever make the `RangeFilter` ive been talking about for a while now I can have this nicer behavior there.